temporal/api/cloud/nexus/v1/message.proto

```diff
@@ -0,0 +1,122 @@
+    repeated IncomingServiceAuthorizationSpec authorization_specs = 4;
+}
+
+message IncomingServiceAuthorizationSpec {
+    // The policy set on this incoming service authorization spec.
+    // Possible values are: allowed_cloud_namespace
+    string policy_type = 1;
+
+    // A policy spec that allows one caller namespace to access the service.
+    AllowedCloudNamespacePolicySpec allowed_cloud_namespace_policy_spec = 2;
+}
+
+message AllowedCloudNamespacePolicySpec {
+    // The namespace that is allowed to call into this service. Calling namespace must be in same account as incoming service.
+    string namespace = 1;
+}
+
+// A binding to a cloud namespace and task queue for dispatching incoming Nexus requests.
+message IncomingService {
+    // The id of the incoming service. This is generated by the server and is immutable.
+    string id = 1;
+
+    // The current version of the incoming service specification.
+    // The next update operation must include this version.
+    string resource_version = 2;
+
+    // The incoming service specification.
+    IncomingServiceSpec spec = 3;
+
+    // The current state of the service.
+    // Possible values: activating, activationfailed, active, updating, updatefailed, deleting, deletefailed, deleted
+    // For any failed state, reach out to Temporal Cloud support for remediation.
+    string state = 4;
+
+    // The id of the async operation that is creating/updating/deleting the service, if any.
+    string async_operation_id = 5;
+
+    // The date and time when the service was created.
+    google.protobuf.Timestamp created_time = 6;
+
+    // The date and time when the service was last modified.
+    // Will not be set if the service has never been modified.
+    google.protobuf.Timestamp last_modified_time = 7;
+}
+
+message OutgoingServiceSpec {
+    // The name of the outgoing service. Must be unique within the namespace.
+    // The name must match `[a-zA-Z_][a-zA-Z0-9_]*`.
+    // This name is immutable.
+    string name = 1;
+
+    // The target that the outgoing service will connect to.
+    TargetSpec spec = 2;
+}
+
+message TargetSpec {
+    // The type of target that is set in this spec.
+    // Possible values are: cloud_incoming_service
+    string target_type = 1;
+    
+    // A target spec for an incoming service within the same account as the outgoing service.
+    TargetCloudIncomingServiceSpec target_cloud_incoming_service_spec = 2;
+}
+
+message TargetCloudIncomingServiceSpec {
+    // The ID of the Incoming Service that the outgoing service will connect to.
+    string incoming_service_id = 1;
+}
+
+// A per-namespace binding from service name to destination service to invoke Nexus requests
+// that are initiated by workflows.
+message OutgoingService {
+    // The name of the outgoing service.
+    string name = 1;
+
+    // The namespace this outgoing service is attached to.
+    string namespace = 2;
```

Comment:
Should this be `= 1`? It's essentially the first part of the identifier.

Reply:
good point. reordered